app/packet_handlers.py

```diff
@@ -529,6 +554,200 @@ async def create_match_handler(session: "Session", packet_data: bytes):
+            removed_session_id = await channel_members.remove(
+                old_match_channel["channel_id"],
+                session["session_id"],
+            )
+            if removed_session_id is not None:
+                old_match_channel_members.remove(removed_session_id)
+
+        for other_session_id in old_match_channel_members:
+            await packet_bundles.enqueue(
+                other_session_id,
+                packets.write_channel_info_packet(
+                    "#multiplayer",
+                    old_match_channel["topic"],
+                    len(old_match_channel_members) - 1
+                    if len(old_match_channel_members) > 0
+                    else 0,
+                ),
+            )
+
+        await packet_bundles.enqueue(
+            session["session_id"],
+            data=packets.write_dispose_match_packet(old_match["match_id"]),
+        )
+
+        packet_params = (
+            old_match["match_id"],
+            old_match["status"] == MatchStatus.PLAYING,
+            old_match["mods"],
+            old_match["match_name"],
+            old_match["match_password"],
+            old_match["beatmap_name"],
+            old_match["beatmap_id"],
+            old_match["beatmap_md5"],
+            [s["status"] for s in slots],
+            [s["team"] for s in slots],
+            [s["account_id"] for s in slots if s["status"] & 0b01111100 != 0],
+            old_match["host_account_id"],
+            vanilla_game_mode,
+            old_match["win_condition"],
+            old_match["team_type"],
+            old_match["freemods_enabled"],
+            [s["mods"] for s in slots] if old_match["freemods_enabled"] else [],
+            old_match["random_seed"],
+        )
+
+        # send the updated match (with password) to users in the match
+        packet_with_password = packets.write_update_match_packet(
+            *packet_params,
+            should_send_password=True,
+        )
+        # TODO: store session_id in slot so channel doesn't need to be fetched
+        old_match_channel = await channels.fetch_one_by_name(
+            f"#mp_{old_match['match_id']}"
+        )
+        assert old_match_channel is not None
+
+        for other_session_id in await channel_members.members(
+            old_match_channel["channel_id"]
+        ):
+            await packet_bundles.enqueue(
+                other_session_id,
+                packet_with_password,
+            )
+
+        # send the updated match (without password) to users in #lobby
+        packet_without_password = packets.write_update_match_packet(
+            *packet_params,
+            should_send_password=False,
+        )
+
+        for other_session_id in await channel_members.members(
+            lobby_channel["channel_id"]
+        ):
+            await packet_bundles.enqueue(
+                other_session_id,
+                packet_without_password,
+            )
+
```

Comment:
we will likely need this code for match joining (todo; test if clicking a multiplayer link in chat will join another before leaving the first)
```suggestion
```